Synchronization fails on newer versions with a long-running system

### Bug Description

Synchronization failed after mobile app upgraded to the version on the server

### Reproduction Steps

1. Connect the mobile app to https://release-international.sormas.netzlink.com/sormas-ui/ using surveillance officer + contact officer role
2. Synchronize the app 
...


### Expected Behavior

App should synchronize successfully, there should be no red arrow message ""Synchronizaton failed, please try later"

### Affected Area(s)

Android app

### SORMAS Version

1.86.0

### Android version/Browser

A 11

### User Role(s)

Surveillance officer + contact officer

### Server URL

https://release-international.sormas.netzlink.com/sormas-ui/

### Additional Information

tablet log
[android-debug.log](https://github.com/SORMAS-Foundation/SORMAS-Project/files/12130060/android-debug.log)

This is very likely caused by https://github.com/SORMAS-Foundation/SORMAS-Project/issues/10721 which has replaced the `DASHBOARD` feature type with individual feature types for the different dashboards, but has failed to apply these changes to the Android database. Because of that, Android apps will still have the `DASHBOARD` feature type in their database, but there's no longer an enum value that can be mapped to it, which leads to this error.

We'll have to fix this by adding a database script that explicitely removes the featureconfiguration entry with the `DASHBOARD` feature type IF IT EXISTS (because it won't on apps that have been installed since). Apps that aren't able to synchronize right now should then be able to do so again after updating to the newest version.

Relevant log output:
```
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: Error trying to synchronizing data in mode 'CompleteAndRepull'
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: de.symeda.sormas.app.backend.common.DaoException: Cannot get enum value of 'DASHBOARD' for field FieldType:name=featureType,class=FeatureConfiguration
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AbstractAdoDao.callBatchTasks(AbstractAdoDao.java:1343)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AbstractAdoDao.deleteInvalid(AbstractAdoDao.java:1041)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.rest.SynchronizeDataAsync.pullMissingAndDeleteInvalidInfrastructure(SynchronizeDataAsync.java:914)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.rest.SynchronizeDataAsync.doInBackground(SynchronizeDataAsync.java:203)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.rest.SynchronizeDataAsync$1.doInBackground(SynchronizeDataAsync.java:984)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at android.os.AsyncTask$3.call(AsyncTask.java:394)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at java.lang.Thread.run(Thread.java:923)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: Caused by: java.sql.SQLException: Cannot get enum value of 'DASHBOARD' for field FieldType:name=featureType,class=FeatureConfiguration
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.field.types.BaseEnumType.enumVal(BaseEnumType.java:29)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.field.types.EnumStringType.sqlArgToJava(EnumStringType.java:55)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.field.BaseFieldConverter.resultToJava(BaseFieldConverter.java:30)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.field.FieldType.resultToJava(FieldType.java:863)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.mapped.BaseMappedQuery.mapRow(BaseMappedQuery.java:62)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.SelectIterator.getCurrent(SelectIterator.java:290)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.SelectIterator.nextThrow(SelectIterator.java:163)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.StatementExecutor.query(StatementExecutor.java:207)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.dao.BaseDaoImpl.query(BaseDaoImpl.java:286)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.QueryBuilder.query(QueryBuilder.java:384)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AbstractAdoDao.lambda$deleteInvalid$2(AbstractAdoDao.java:1044)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AbstractAdoDao.$r8$lambda$NTKb-Pa5x25lA-Rygl3W8yCWaxk(AbstractAdoDao.java:0)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AbstractAdoDao$$ExternalSyntheticLambda0.call(R8$$SyntheticClass:0)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.misc.TransactionManager.callInTransaction(TransactionManager.java:238)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.StatementExecutor.doCallBatchTasks(StatementExecutor.java:639)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.StatementExecutor.doCallBatchTasks(StatementExecutor.java:623)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.stmt.StatementExecutor.callBatchTasks(StatementExecutor.java:603)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.dao.BaseDaoImpl.callBatchTasks(BaseDaoImpl.java:734)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AbstractAdoDao.callBatchTasks(AbstractAdoDao.java:1341)
07-21 16:50:05.070  4816  6609 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	... 9 more
```

--- a/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java
+++ b/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java
@@ -188,7 +188,7 @@ public class DatabaseHelper extends OrmLiteSqliteOpenHelper {
 	public static final String DATABASE_NAME = "sormas.db";
 	// any time you make changes to your database objects, you may have to increase the database version
 
-	public static final int DATABASE_VERSION = 346;
+	public static final int DATABASE_VERSION = 347;
 
 	private static DatabaseHelper instance = null;
 
@@ -3080,6 +3080,10 @@ public class DatabaseHelper extends OrmLiteSqliteOpenHelper {
 						+ "snapshot, reportDate, reportingUser_id, environmentName, description, externalId, responsibleUser_id, investigationStatus, environmentMedia, "
 						+ "waterType, otherWaterType, infrastructureDetails, otherInfrastructureDetails, waterUse, otherWaterUse, location_id, id FROM tmp_environments");
 				getDao(Environment.class).executeRaw("DROP TABLE tmp_environments");
+
+			case 346:
+				currentVersion = 346;
+				getDao(FeatureConfiguration.class).executeRaw("DELETE FROM featureConfiguration WHERE featureType = 'DASHBOARD';");
 
 				// ATTENTION: break should only be done after last version
 				break;

